app/lib/actions/pulp3/orchestration/repository/upload_content.rb

```diff
@@ -4,28 +4,58 @@ module Orchestration
       module Repository
         class UploadContent < Pulp3::Abstract
           include Actions::Helpers::OutputPropagator
-          def plan(repository, smart_proxy, file, unit_type_id)
+
+          # rubocop:disable Metrics/MethodLength
+          # rubocop:disable Metrics/CyclomaticComplexity
+          def plan(repository, smart_proxy, file, unit_type_id, options)
             sequence do
               checksum = Digest::SHA256.hexdigest(File.read(file[:path]))
               duplicate_sha_path_content_list = ::Katello::Pulp3::PulpContentUnit.find_duplicate_unit(repository, unit_type_id, file, checksum)
               duplicate_content_href = duplicate_sha_path_content_list&.results&.first&.pulp_href
 
               unless duplicate_content_href
-                duplicate_sha_artifact_list = ::Katello::Pulp3::Api::Core.new(smart_proxy).artifacts_api.list("sha256": Digest::SHA256.hexdigest(File.read(file[:path])))
+                if unit_type_id == 'ostree_ref'
+                  duplicate_sha_artifact_list = ::Katello::Pulp3::Api::Core.new(smart_proxy).artifacts_api.list(
+                    'checksum': Digest::SHA256.hexdigest(File.read(file[:path])))
+                else
+                  duplicate_sha_artifact_list = ::Katello::Pulp3::Api::Core.new(smart_proxy).artifacts_api.list(
+                    'sha256': Digest::SHA256.hexdigest(File.read(file[:path])))
```

Comment:
does the artifacts api really support checksum and sha256 arguments?  Looking at the client bindings ruby (pulpcore 3.15.2), i don't see checksum listed